logdna/logdna.py

```diff
@@ -39,57 +42,66 @@ def __init__(self, key, options={}):
         self.index_meta = options.get('index_meta', False)
         self.flush_limit = options.get('flush_limit', defaults['FLUSH_BYTE_LIMIT'])
         self.flush_interval = options.get('flush_interval', defaults['FLUSH_INTERVAL'])
+        self.retry_interval_secs = options.get('retry_interval_secs', defaults['RETRY_INTERVAL_SECS'])
         self.tags = options.get('tags', [])
+        self.buf_retention_limit = options.get('buf_retention_limit', defaults['BUF_RETENTION_LIMIT'])
+        self.buf_size = 0
+        self.exception_flag = False
 
         if isinstance(self.tags, str):
             self.tags = [tag.strip() for tag in self.tags.split(',')]
         elif not isinstance(self.tags, list):
             self.tags = []
-
-
         self.setLevel(logging.DEBUG)
         self.lock = threading.RLock()
 
-        self.session = requests.Session()
-        retry = Retry(connect=self.connection_retries, backoff_factor=self.retry_backoff_factor)
-        adapter = HTTPAdapter(max_retries=retry)
-        self.session.mount('https://', adapter)
+    def get_message_size(self, message):
```

Comment:
See my main comment - I think we should just continue to check the size of `self.buf` directly so this function is not necessary. But a couple Python notes:
1/ This doesn't need to (and shouldn't be) a class method.
2/ To me it feels more Python-y to use `reduce` instead of having this function, like `reduce(lambda x, key: x + sys.getsizeof(key) + sys.getsizeof(message[key]), message, 0)`